File: services/usersPy/update.py
from libs.responseLib import success, failure;
from libs.dataExtraction import extractUserSubFromEvent;
import boto3;
from os import environ;
from datetime import datetime
dynamodb = boto3.resource('dynamodb', region_name=environ.get('REGION'));
table = dynamodb.Table(environ.get('tableName'));
def handler(event, context):
    try:
        _event = json.load(event)
    except Exception as e:
        _event = event
    userId = _event['pathParameters']['userId'] if _event['pathParameters'] and _event['pathParameters']['userId'] else extractUserSubFromEvent(_event)
    try:
        key={
            'pK': f"USER#{userId}",
            'sK': "USER"
        }
        updateExpression = f`SET \
        {'userName = :userName,' if data.userName else ''}\
        {'userMail = :userMail,' if data.userMail else ''}\
        {'userPhone = :userPhone,' if data.userPhone else ''}\
        updatedAt = :updateAt`
        expressionAttributeValues = {
          **({":userName": data.userName} if data.userName else {}),
          **({":userMail": data.userMail} if data.userMail else {}),
          **({":userPhone": data.userPhone} if data.userPhone else {}),
          ":updatedAt": int(datetime.now().timestamp()*1000)
        }
        updateQuery = table.update_item(
            Key=key,
            UpdateExpression=updateExpression,
            ExpressionAttributeValues=expressionAttributeValues,
            ReturnValues="ALL_NEW"
        )
        userUpdatedAttributes = updateQuery['Attributes']
        return success(userUpdatedAttributes);
    except Exception as e:
        return failure({'message': str(e)});

File: services/userJS/update.js
import { success, failure } from "../../libs/responseLib";
import { extractIdentityIdFromEvent } from "../../libs/dataExtraction";
import dynamoDb from "../../libs/dynamodb-lib";

export async function handler(event, context) {
  try {
    const data = typeof event.body === typeof "string" ? JSON.parse(event.body):event.body;
    const userId = data.userId || extractIdentityIdFromEvent(event);
    const updateExpression = `SET ${data.userName ? 'userName = :userName,' : ''}\
      ${data.userMail ? ' userMail = :userMail,' : ''}\
      ${data.userPhone ? ' userPhone = :userPhone,' : ''}\
      updatedAt = :updateAt`;
    const expressionAttributeValues = {
      ...(data.userName ? {":userName": data.userName}:{}),
      ...(data.userMail ? {":userMail": data.userMail}:{}),
      ...(data.userPhone ? {":userPhone": data.userPhone}:{}),
      ":updatedAt": Date.now()
    }
    const params = {
      TableName: process.env.tableName,
      Key: {
        pK: `USER#${userId}`,
        sK: 'USER',
      },
      UpdateExpression: updateExpression,
      ExpressionAttributeValues: expressionAttributeValues,
      ReturnValues: "ALL_NEW"
    };
    const updateQuery = await dynamoDb.update(params);
    const userUpdatedAttributes = updateQuery.Attributes;
    return success(userUpdatedAttributes);
  } catch (e) {
    return failure({
      message: "Error updating user",
      error: e.message
    });
  }
}
